src/tasks/barfTasks.ts

```diff
@@ -264,76 +259,82 @@ const turns: AdventureAction[] = [
           Macro.externalIf(underwater, Macro.item($item`pulled green taffy`)).meatKill()
         ).abort()
       );
-      return get("_sourceTerminalDigitizeMonsterCount") !== start;
+      return;
     },
     spendsTurn: () => !SourceTerminal.getDigitizeMonster()?.attributes.includes("FREE"),
     sobriety: Sobriety.EITHER,
+    completed: () => get("_sourceTerminalDigitizeMonsterCount") !== digitize,
   },
   {
     name: "Guaranteed Kramco",
-    available: () => kramcoGuaranteed(),
-    execute: () => {
-      freeFightPrep(new Requirement([], { forceEquip: $items`Kramco Sausage-o-Matic™` }));
+    ready: () => kramcoGuaranteed(),
+    do: (): void => {
+      prep("free", {
+        requirements: new Requirement([], { forceEquip: $items`Kramco Sausage-o-Matic™` }),
+      });
       adventureMacroAuto(drunkSafeWander("wanderer"), Macro.basicCombat());
-      return !kramcoGuaranteed();
     },
+    completed: () => !kramcoGuaranteed(),
```

Comment:
How often is our completed condition the negation of our ready condition? Should there be a shortcut for this?